### PR TITLE
ios color asset support for runtime dark & light color switching 

### PR DESCRIPTION
**Pull request by jittya**
To support runtime theme switch the generated color assets files need to be copied to iosApp. which can be automated by a gradle task like below and configure it as iOS  run script.

val copyColorAssetsToIOSApp = tasks.register<Copy>("copyColorAssetsToIOSApp") {
    from("$rootDir/resources/build/generated/moko/iosMain/res/Assets.xcassets")
    into("$rootDir/iosApp/iosApp/Assets.xcassets/colors")
}

Now to get the color from the iOS app assets we need to use UIColor.colorNamed(“colorName”) function. Below extension method can be used for that

fun ColorResource.getThemeColor(): UIColor {
   return UIColor.colorNamed(this.name)!!
}

### CODE REVIEW

**Comment by benMohamed** [comment]
Can you please accept this ?

File: resources/src/iosMain/kotlin/dev/icerock/moko/resources/ColorResourceExt.kt [hunk 1]
```diff
@@ -22,3 +24,15 @@ fun ColorResource.getColor(userInterfaceStyle: UIUserInterfaceStyle): Color {
...
+/**
+ * Returns null if no color asset is found in the ios app.
+ * update and configure copyColorAssetsToIOSApp in your ios run script to automate the copy process.
+ *       val copyColorAssetsToIOSApp = tasks.register<Copy>("copyColorAssetsToIOSApp") {
+ *              from("$rootDir/resources/build/generated/moko/iosMain/res/Assets.xcassets")
+ *              into("$rootDir/iosApp/iosApp/Assets.xcassets/colors")
+ *       }
+ */
+fun ColorResource.getThemeColor(): UIColor {
+    return UIColor.colorNamed(this.name)!!
```
Comment (Alex009) [inline resources/src/iosMain/kotlin/dev/icerock/moko/resources/ColorResourceExt.kt:37]: need to use `UIColor.colorNamed(this.name, inBundle = this.bundle)` - with it no any additional gradle tasks require

File: resources/src/commonMain/kotlin/dev/icerock/moko/resources/ColorResource.kt [hunk 2]
```diff
@@ -6,8 +6,8 @@ package dev.icerock.moko.resources
 
 import dev.icerock.moko.graphics.Color
 
-sealed class ColorResource {
-    class Single(val color: Color) : ColorResource()
+sealed class ColorResource(val name: String) {
```
Comment (Alex009) [inline resources/src/commonMain/kotlin/dev/icerock/moko/resources/ColorResource.kt:9]: name have no sense for not apple targets. we should change `ColorResource` to expect/actual. for apple target we should have `name` and `bundle` args.
for android we should use resource arg, with generation of xml colors file.
for other targets - save current way with manual selection

File: resources-test/src/commonMain/kotlin/dev/icerock/moko/resources/test/createColorResourceMock.kt [hunk 3]
```diff
@@ -7,4 +7,4 @@ package dev.icerock.moko.resources.test
 import dev.icerock.moko.graphics.Color
 import dev.icerock.moko.resources.ColorResource
 
-fun createColorResourceMock(): ColorResource = ColorResource.Single(Color(0, 0, 0, 0))
+fun createColorResourceMock(): ColorResource = ColorResource.Single(Color(0, 0, 0, 0), "black")
```
Comment (Alex009) [inline resources-test/src/commonMain/kotlin/dev/icerock/moko/resources/test/createColorResourceMock.kt:10]: it's apple-specific details, should be removed

File: resources-generator/src/main/kotlin/dev/icerock/gradle/generator/ColorsGenerator.kt [hunk 4]
```diff
@@ -72,9 +72,9 @@ abstract class ColorsGenerator(
     protected open fun getPropertyModifiers(): Array<KModifier> = emptyArray()
     protected open fun getPropertyInitializer(color: ColorNode): CodeBlock? {
         val className = if (color.isThemed()) {
-            "Themed(light = Color(0x${color.lightColor}), dark = Color(0x${color.darkColor}))"
+            "Themed(light = Color(0x${color.lightColor}), dark = Color(0x${color.darkColor}), name = \"${color.name}\")"
         } else {
-            "Single(color = Color(0x${color.singleColor}))"
+            "Single(color = Color(0x${color.singleColor}), name = \"${color.name}\")"
```
Comment (Alex009) [inline resources-generator/src/main/kotlin/dev/icerock/gradle/generator/ColorsGenerator.kt:77]: here we should improve logic to support apple & android native way of colors usage. i describe it in `ColorResource` file comment

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: resources/src/iosMain/kotlin/dev/icerock/moko/resources/ColorResourceExt.kt (hunk 1 above)
Comment (Alex009) [reply resources/src/iosMain/kotlin/dev/icerock/moko/resources/ColorResourceExt.kt:37]: fixed in #439

File: resources/src/commonMain/kotlin/dev/icerock/moko/resources/ColorResource.kt (hunk 2 above)
Comment (Alex009) [reply resources/src/commonMain/kotlin/dev/icerock/moko/resources/ColorResource.kt:9]: fixed in #439

File: resources-test/src/commonMain/kotlin/dev/icerock/moko/resources/test/createColorResourceMock.kt (hunk 3 above)
Comment (Alex009) [reply resources-test/src/commonMain/kotlin/dev/icerock/moko/resources/test/createColorResourceMock.kt:10]: fixed in #439

File: resources-generator/src/main/kotlin/dev/icerock/gradle/generator/ColorsGenerator.kt (hunk 4 above)
Comment (Alex009) [reply resources-generator/src/main/kotlin/dev/icerock/gradle/generator/ColorsGenerator.kt:77]: fixed in #439

**Pull request merged**